api/PayrollProcessor.Core.Domain/Intrastructure/Operations/Commands/ICommandDispatcher.cs

```diff
@@ -0,0 +1,12 @@
+using System.Threading;
+using LanguageExt;
+
+namespace PayrollProcessor.Core.Domain.Intrastructure.Operations.Commands
+{
+    public interface ICommandDispatcher
+    {
+        TryOptionAsync<Unit> Dispatch(ICommand command, CancellationToken token = default);
```

Comment:
`TryOptionAsync<T>` requires a generic type parameter because in functional programming `void` is not a real return type.

`LanguageExt` provides `Unit` which is a return type that implies no value but side effects did occur.

It would be really interesting to also use this approach on the client side - instead of a `: void` return type on Akita service methods, we could use a `: Unit` return type using something like https://gcanti.github.io/fp-ts/